sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -123,6 +173,19 @@ case class InSet(child: Expression, hset: Set[Any])
   override def eval(input: InternalRow): Any = {
     hset.contains(child.eval(input))
   }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val hsetName = ctx.freshName("InSet")
+    InSet.addHSet(hsetName, hset)
```

Comment:
as commented on the other pr, we can just pass this in by generalizing the references to include any arbitrary object.